src/coord/src/catalog.rs

```diff
@@ -623,7 +631,19 @@ impl Catalog {
             events.push(catalog.insert_item(id, oid, name, item));
         }
 
-        Ok((catalog, events))
+        let catalog_content_version = catalog.storage().get_catalog_content_version()?;
+
+        if CATALOG_CONTENT_MIGRATIONS.len() > catalog_content_version {
```

Comment:
So are we currently on `catalog_content_version` 0 and we'll update each time we have to add a migration? 

Also, I don't believe this currently handles the situation in which a user's catalog is multiple versions behind.

Reply:
We are currently on _NULL_ version, and the first time you start Materialize after this commit lands, you will be on version 0, and then see that there is a pending `CATALOG_CONTENT_MIGRATION` and run that.

This should handle being multiple versions behind as long as the catalog is persistent, i.e. the call to `Catalog::open` is tail recursive and the terminal condition for recursion is that you have seen all of the migrations listed. Let me know if that explanations jives with you or if I've overlooked something.